fdbclient/NativeAPI.actor.cpp

```diff
@@ -255,22 +255,23 @@ void DatabaseContext::getLatestCommitVersions(const Reference<LocationInfo>& loc
 
 	std::map<Version, std::set<Tag>> versionMap; // order the versions to be returned
 	for (int i = 0; i < locationInfo->locations()->size(); i++) {
-		int updatedVersionMap = 0;
+		bool updatedVersionMap = false;
 		Version commitVersion = invalidVersion;
 		Tag tag = invalidTag;
-		UID uid = locationInfo->locations()->getId(i);
-		if (ssidTagMapping.find(uid) != ssidTagMapping.end()) {
-			tag = ssidTagMapping[uid];
+		auto iter = ssidTagMapping.find(locationInfo->locations()->getId(i));
+		if (iter != ssidTagMapping.end()) {
+			tag = iter->second;
 			if (ssVersionVectorCache.hasVersion(tag)) {
 				commitVersion = ssVersionVectorCache.getVersion(tag); // latest commit version
 				if (commitVersion < readVersion) {
+					updatedVersionMap = true;
 					versionMap[commitVersion].insert(tag);
 				}
 			}
 		}
 		if (!updatedVersionMap) {
 			TraceEvent(SevDebug, "CommitVersionNotFoundForSS")
-			    .detail("InSSIDMap", ssidTagMapping.find(uid) != ssidTagMapping.end() ? 1 : 0)
+			    .detail("InSSIDMap", iter != ssidTagMapping.end() ? 1 : 0)
```

Comment:
I am concerned about the overhead that would be imposed by this event - Is this trace event really needed? What additional actionable information does this provide in production?

Reply:
It is in `SevDebug`, so off by default. I think it is useful to have if we see a problem in `ckperf` that is hard to reproduce elsewhere. In that case we may wish to change the trace severity knob to Debug level. The event will tell us which of the 3 conditionals is failing.

Reply:
Ok. Would you like to log the contents of the version vector too (this may help with identifying special scenarios like post-recovery at which point the version vector is going to be empty)?